.buildkite/pipeline.elastic-agent-binary-dra.yml

```diff
@@ -13,17 +13,32 @@ steps:
     # don't run snapshot builds with prereleases (non empty VERSION_QUALIFIER) unless forced (RUN_SNAPSHOT=true)
     if: build.env("RUN_SNAPSHOT") == "true" || (build.env('VERSION_QUALIFIER') == null && (build.branch == 'main' || build.branch =~ /^[0-9]+\.[0-9x]+\$/))
     steps:
-    - label: ":package: Build Elastic-Agent Core Snapshot"
+    - label: ":package: Elastic-Agent Core Snapshot"
       commands:
         - .buildkite/scripts/steps/build-agent-core.sh
-      key: "build-dra-snapshot"
+      key: "build-dra-snapshot-x86"
       artifact_paths:
         - "build/distributions/**/*"
       agents:
         provider: "gcp"
         machineType: "c2-standard-16"
       env:
         DRA_WORKFLOW: "snapshot"
+        PLATFORMS: "linux/amd64,windows/amd64,darwin/amd64,darwin/arm64"
```

Comment:
I think it is easier to read with spaces than with `commas`, and it follows the same pattern as done in other projects.

Shall we use the same format as done in https://github.com/elastic/beats/blob/2c6e5affa25bd29bda0e56d506074967cda28f7d/.buildkite/metricbeat/pipeline.yml#L402

Reply:
Not sure about readability but the PLATFORMS list can be separated by `,`, `;` or space so the expressions are equivalent. 
See https://github.com/elastic/elastic-agent/blob/31aac31e6f5aa63932aa37eb1f30cbeab00e22bf/dev-tools/mage/platforms.go#L281 and https://github.com/elastic/elastic-agent/blob/31aac31e6f5aa63932aa37eb1f30cbeab00e22bf/dev-tools/mage/common.go#L505-L507

If we want to keep pipelines under engineering productivity consistent by using a space-separated PLATFORMS list it's fine by me.

Be careful as this feature does not apply to all env vars containing a list (see PACKAGES parsing for example) https://github.com/elastic/elastic-agent/blob/b20efc1801b1b18d9e0b035cea34ff1b942a9066/dev-tools/mage/crossbuild.go#L46-L56

Reply:
That's good to know! Thanks for the details.

I'm more inclined to use the same patterns for obvious reasons: managing `> 200 GitHub repositories` is challenging, and we can only sustain by following similar practices everywhere. We agree that exceptions make things complicated.

By the way, I'm not blocking this, but if possible, I'd encourage everyone to follow those practices sooner than latter.